R/CreatePackageVignette.R

```diff
@@ -0,0 +1,101 @@
+#' @title pkgnet Report as Vignette
+#' @name CreatePackageVignette
+#' @description Create pkgnet package report as an Rmarkdown vignette. This
+#'    vignette is able to be rendered with the
+#'    \code{\link[knitr:vignette_engines]{knitr::rmarkdown}} vignette engine
+#'    into HTML vignettes upon package building. It is also compatible with
+#'    \code{\link[pkgdown:build_articles]{pkgdown}} sites.
+#' @param pkg_name (string) name of a package
+#' @param pkg_path (string) The path to the package repository. If given, coverage
+#'                 will be calculated for each function. \code{pkg_path} can be an
+#'                 absolute or relative path.
+#' @param pkg_reporters (list) a list of initialized package reporters
+#' @param vignette_path (string) The path and filename of the output vignette
+#'    file. The default assumes your working directory is the package root.
+#' @importFrom rlang enexpr
+#' @importFrom assertthat assert_that is.string is.readable
+#' @importFrom tools file_ext
+#' @importFrom R6 is.R6Class
+#' @importFrom glue glue
+#' @export
+CreatePackageVignette <- function(pkg_name
+                                  , pkg_path = NULL
+                                  , pkg_reporters = list(
```

Comment:
Why not `DefaultReporters()`?

Reply:
I didn't want to include `SummaryReporter`. It felt unnecessary to print the whole DESCRIPTION file of a package in a vignette. 

This situation actually makes me feel like `DefaultReporters()` isn't that useful. There isn't any use for it _besides_ of `CreatePackageReport()` and we can do the same thing for the `CreatePackageReport` default.

Reply:
Good point about the DESCRIPTION file being redundant for a vignette. Let's mull over `DefaultReporters()` fate in a later PR.